### PR TITLE
Update Exercise 6.5 to not use helper functions

### PR DESCRIPTION
**Pull request by sxs9174**
For swapfl assumption is made that empty list and list with one
element will return false given they do not have first and last
element. This assumption is based on strict interpretation of 
question that it needs to compare the first and last elements.
On this assumption base case is made with two elements
swapped, and inductive case with head and tail being different
but elements in between are same.

This allows to define swapfl in terms of itself, without any extra
helper functions.

### CODE REVIEW

File: chapter-6/exercises.pl [hunk 1]
```diff
@@ -100,37 +100,13 @@
...
-swapfl(List1,List2) :-
-  compareFirstLast(List1,List2),
-  compareFirstLast(List2,List1),
-  chopOffEnds(List1, Outlist1),
-  chopOffEnds(List2, Outlist2),
-  equal(Outlist1,Outlist2).
-
-compareFirstLast([H|T1],L2) :- rev(L2, [H|_]).
+%% As per strict interpretation of the question. Empty lists or lists with one
+%% element cannot be compared since they are missing the first and last element.
```
Comment (lolotobg) [inline chapter-6/exercises.pl]: The first and last elements of the list with length 1 `[a]` are both `a`, so the predicate `swapfl([a], [a])` should be true, at least the way I read the task's description. The case with zero length lists is not so clear though.
P.s. Bare in mind that if you add a `swapfl([X], [X])` clause, the second rule will start to (wrongly) accept the case with 2 equal 2-element lists (`swapfl([a, b], [a, b]).` will be true).

Comment (sxs9174) [inline chapter-6/exercises.pl]: As per my interpretation of lists in Prolog and other languages a list with one element is basically the element cons with empty list. That is [a] would a :: [] or something like cons(a, []). So in this case, the first element of a list with single element would always be the head element and last element or tail element would be the empty list []. A list with single element like [a] as per my understanding cannot be represented as [] :: a.
In short if you want to divide a list with single element in two parts, then you can do so. Also those two parts cannot be interchanged as per definition. That is an empty list cannot be first element in a list, unless it is a list of list.

However you are right. If the task description mentions that lists with single element are supposed to be taken into account, then the above solution would fall apart. The solution is not generic enough.

**Comment by sxs9174** [comment]
Closing the pull request as the given solution is not optimal; and not generic enough.
